kotlin-utils/src/jsMain/kotlin/org/modelix/kotlin/utils/ContextValue.js.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.modelix.kotlin.utils
+
+import kotlinx.coroutines.currentCoroutineContext
+import kotlinx.coroutines.withContext
+import kotlin.coroutines.AbstractCoroutineContextElement
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.ContinuationInterceptor
+import kotlin.coroutines.CoroutineContext
+
+actual class ContextValue<E> {
+    private val initialStack: List<E>
+    private var synchronousValueStack: List<E>? = null
+    private var stackFromCoroutine: List<E>? = null
+    private val contextElementKey = object : CoroutineContext.Key<ContextValueElement<E>> {}
+    private var isInSynchronousBlock = false
```

Comment:
I am not sure about the JS backend and the expected access pattern here, but are you sure this doesn't need a `@Volatile` annotation?

Reply:
There is no `@Volatile` in JS.